src/architecture/serialization.md

```diff
@@ -1,533 +1,193 @@
-| `message_id`       | `u16`             | 2..4          | number     |
-| `payload_length`   | `u32`             | 6..10         | -          |
-| `body`             | `&[u8]`           | 10..-64       | object     |
-| `signature`        | Ed25519 signature | -64..         | hex string |
-
-!!! tip
-    For the binary format, the table uses the type notation taken
-    from [Rust][rust]. Offsets also correspond to [the slicing syntax][rust-slice],
-    with the exception that Rust does not support negative offsets,
-    which denote an offset relative to the end of the byte buffer.
-
-### Padding
-
-The first byte of the message is used as padding. It must be equal to `0`.
-
-### Protocol Version
-
-The major version of the Exonum serialization protocol. Currently, `0`.
-
-**Binary presentation:** `u8` (unsigned 1-byte integer).  
-**JSON presentation:** number.
+Besides the description of structures, the `.rs` files also contain additional
+code and functions required for protobuf serialization and deserialization.
 
-### Service ID
+For convenience, the [protobuf descriptions][proto-files] of the typical
+structures used in Exonum are already included in the framework.
 
-Used when the message is a transaction.
-Sets the [service](services.md) that a transaction belongs to.
-The pair `(service_id, message_id)` is
-used to look up the implementation of [the transaction interface](transactions.md#interface)
-(e.g., `verify` and `execute` methods).
+## Building Exonum with Protobuf Serialization
 
-**Binary presentation:** `u16` (unsigned 2-byte integer).  
-**JSON presentation:** number.
+Exonum includes the
+[`exonum_build`](https://github.com/exonum/exonum/tree/master/exonum_build)
+crate which lets users add the `protobuf_generate` function to their
+`build.rs`. This function automatically generates the `.rs` files for all the
+`.proto` files during the build process. To use `protobuf_generate`, add the
+following code to `build.rs` indicating the folder which contains the `.proto`
+files:
 
-### Message ID
-
-`message_id` defines the type of message within the service.
-
-!!! note "Example"
-    [The sample cryptocurrency service][cryptocurrency] includes 2 main
-    types of transactions: `TxCreateWallet` for creating a wallet with an initial
-    coins balance,
-    and `TxTransfer` for coin transfer.
-
-**Binary presentation:** `u16` (unsigned 2-byte integer).  
-**JSON presentation:** number.
+```rust
+use exonum_build::protobuf_generate;
 
-### Payload length
+protobuf_generate("src/proto", &["src/proto"], "example_mod.rs")
+```
 
-The length of the entire message serialization (including its header and signature).
+To use protobuf-generated Rust structures, users first need to create a module
+which will include the protobuf-generated files:
 
-**Binary presentation:** `u32` (unsigned 4-byte integer).  
-**JSON presentation:** (not serialized).
+```rust
+extern crate exonum;
```

Comment:
Why is `extern crate exonum` included here, in the container module? I'd just leave it out. In the 2018 edition, it's not required at all, and in 2015 edition it should be placed in `lib.rs` (otherwise, we would need to write `use self::exonum::...` later).